api/models/cartDao.js

```diff
@@ -0,0 +1,118 @@
+const dataSource = require('./dataSource');
+
+const cartProductQuantity = async(userId, productId, productOptionId) => {
+  try {
+    const [result] = await dataSource.query(
+      `
+      SELECT SUM(quantity) as sum FROM carts 
+      where user_id = ? and product_id = ? and product_option_id = ?
+      `,
+      [userId, productId, productOptionId]
+    )
+    return parseInt(result.sum, 10)
+  } catch (err) {
+    const error = new Error('isincart error');
+    error.statusCode = 400;
+    throw error;
+  }
+
+}
+
+const deleteCartItem = async(userId, productId, productOptionId) => {
+  try {
+    await dataSource.query(
+      `
+      DELETE FROM carts
+      where user_id = ? and product_id = ? and product_option_id = ?
+      `,
+      [userId, productId, productOptionId]
+    )
+  } catch (err) {
+    const error = new Error('INVALID_DATA_INPUT');
+    error.statusCode = 400;
+    throw error;
+  }
+}
+
+
+const checkInventory = async(productId, productOptionId) => {
+  try {
+    const [product] = await dataSource.query(
+      `
+      SELECT * FROM product_options
+      where id = ? and product_id = ?
+      `,
+      [productOptionId, productId]
+    )
+    return product.inventory;
+  } catch (err) {
+    const error = new Error('INVALID_DATA_INPUT');
+    error.statusCode = 400;
+    throw error;
+  }
+}
+
+const updateInventory = async(productId, productOptionId, updatedInventory) => {
+  try {
+    await dataSource.query(
+      `
+      UPDATE product_options
+      SET inventory = inventory - ?
+      where id = ? and product_id = ?
+      `,
+      [updatedInventory, productOptionId, productId]
+    )
+
+  } catch (err) {
+    const error = new Error('INVALID_DATA_INPUT');
+    error.statusCode = 400;
+    throw error;
+  }
+}
+
+const addProduct = async(userId, productId, productOptionId, quantity) => {
```

Comment:
함수의 이름이 장바구니를 추가 한다는 느낌보다 제품을 데이터를 추가하는 느낌이 더 강합니다. 함수 이름을 아래의 예시로 바꾸면 좋을 것 같습니다.
```suggestion
const addCart = async(userId, productId, productOptionId, quantity) => {
```
or

```suggestion
const createCart = async(userId, productId, productOptionId, quantity) => {
```